### ISSUE TITLE
aria-required-children should not permit mixing child roles

### ISSUE THREAD

**Issue by WilcoFiers** [dequelabs/axe-core#3468]
Axe-core's aria-required-children rule shouldn't pass the following:

```html
<div role="list">
  <li>Item 1</li>
  <span role="link">Item 2</span>
</div>
```

WAI-ARIA is a little vague about whether or not you're allowed to do this. There are scenarios, such as for radiogroup where owned elements might not all need to be of the required owned elements type, but for something like lists and trees, that doesn't seem right.

Failing the above example is required for consistency with the [ARIA required owned elements](https://www.w3.org/WAI/standards-guidelines/act/rules/bc4a75/proposed/) ACT rule.

**Comment by straker** [comment]
We can do this by filtering the `ownedRoles` array and seeing which ones are not required

```js
const invalidRoles = ownedRoles.filter(role => !required.includes(role));
  if (invalidRoles) {
    this.data({

    })
    return false;
  }
```

**Comment by padmavemulapati** [comment]
Validated with the latest develop branch code base, the script below is failing `arai-required-children` rule
```
<div role="list">
    <li>Item 1</li>
    <span role="link">Item 2</span>
  </div>
```
<img width="1446" alt="image" src="https://user-images.githubusercontent.com/47409579/187229819-9bd2052c-17fa-4ee5-bd19-e720f1cbde38.png">
